typeinfo.go

```diff
@@ -348,17 +348,30 @@ func getFields(typ reflect.Type) fields {
 		if name == "-" {
 			continue
 		}
+		ftyp := f.Type
+		if ftyp.Kind() == reflect.Ptr {
+			ftyp = ftyp.Elem()
+		}
+		if opts.Contains("inline") && ftyp.Kind() == reflect.Struct {
```

Comment:
Indeed `embed` would fit better, however there was already precedence and `inline` shows up in couple of places:

- proposed `inline` opt for `encoding/json` (https://github.com/golang/go/issues/6213)
- existing `inline` opt in `gopkg.in/yaml.v2` (https://godoc.org/gopkg.in/yaml.v2#Marshal)

So I think I'd better to stick with `inline`.